stl/inc/mdspan

```diff
@@ -289,15 +295,21 @@ public:
     using rank_type    = typename _Extents::rank_type;
     using layout_type  = layout_left;
 
+    static_assert(
+        _Is_extents<_Extents>, "Extents must be a specialization of extents (N4928 [mdspan.layout.left.overview]/2).");
+    static_assert(_Extents::_Is_index_space_size_representable(),
+        "If Extends::rank_dynamic() == 0 is true, then the size of the multidimensional index space Extents() is "
+        "representable as a value of type typename Extends::index_type.");

```

Comment:
1. This file says "Extends" 6 times total, which appears to be a typo. Additionally, 
2. This appears to quote Standardese, but without citing the paragraph, and 
3. It's phrased in a way that may be confusing to user-programmers. The previous static assertion clearly said "Cats must be cute." but this one says "If condition, then the cat is cute."

Reply:
My next PR will fix `1` and `3`. For now, I'm not going to cite the paragraph for `2`, because current working draft (N4928) does not have this *Mandates* clause (it was part of LWG-3876).